src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2952,7 +2941,7 @@ private bool IsEndOfNameInExplicitInterface()
                     // Use else if, rather than if, because if we see both a constructor initializer and a constraint clause, we're too lost to recover.
                     var colonToken = this.CurrentToken;
                     // Set isStatic to false because pretending we're in a static constructor will just result in more errors.
-                    ConstructorInitializerSyntax initializer = this.ParseConstructorInitializer(identifier.ValueText, isStatic: false);
+                    var initializer = this.ParseConstructorInitializer();
```

Comment:
No, it is not implied, the type name must be explicitly spelled out in the initializer, then it will be implied. For example, "**ArrayBuilder<int>**.GetInstance()".